### PR TITLE
Us 47

### PR DESCRIPTION
**Pull request by davehardy632**
Merchant can delete an item.

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -33,6 +33,6 @@ def item_total_value_for_merchant(merchant_id)
   end
 
   def self.sort_by_status
-    order("status = 0 DESC, status = 1 DESC, status = 2 DESC, status = 3 DESC")
```
Comment (JustinD85) [inline app/models/order.rb:36]: This line was throwing that TLDR error message about deprecation... so I changed it to use a symbol and it works still with no errors. Let me know if I misunderstood the setup here.

File: spec/factories/order_item_factory.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
     order
     item
     sequence(:quantity) { |n| ("#{n}".to_i+1)*2 }
-    sequence(:price) { |n| ("#{n}".to_i+1*1.5) }
+    sequence(:price) { |n| (item && item.price || "#{n}".to_i+1*1.5) }
```
Comment (JustinD85) [inline spec/factories/order_item_factory.rb:6]: this line says, if `order_item` is given and item, then set the price of `order_item` to that items current price else give it a value based on the logic that was already there. This enables us to give `order_item` a item in our tests and it's price is set correctly.

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -33,14 +33,14 @@
...
 
-      @item_1 = create(:item, user: @user)
-      @item_2 = create(:item, user: @user)
-      @item_3 = create(:item, user: @user)
-      @item_4 = create(:item, user: @user)
-      @item_5 = create(:item, user: @user)
-      @item_6 = create(:item, user: @user)
-      @item_7 = create(:item, user: @user)
-      @item_8 = create(:item, user: @user)
+      @item_1 = create(:item, user: @merchant)
```
Comment (JustinD85) [inline spec/models/user_spec.rb:36]: I changed this from user to merchant as only merchants have items. I changed the tests below it also, so I am now not confused when reading it 😄

**Review by JustinD85 — APPROVED** [review]

**Review by MillsProvosty — APPROVED** [review]

**Review by ryanmillergm — APPROVED** [review]

File: app/models/order.rb (hunk 1 above)
Comment (MillsProvosty) [reply app/models/order.rb:36]: We got the same one, and we chose to ignore it. This looks fine.

**Pull request merged**